collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -775,83 +860,37 @@ public List<MetricFamilySamples> collect() {
-                        "jmx_scrape_error",
-                        Type.GAUGE,
-                        "Non-zero if this scrape failed.",
-                        samples));
-        samples = new ArrayList<>();
-        samples.add(
-                new MetricFamilySamples.Sample(
-                        "jmx_scrape_cached_beans",
-                        new ArrayList<>(),
-                        new ArrayList<>(),
-                        stalenessTracker.cachedCount()));
-        mfsList.add(
-                new MetricFamilySamples(
-                        "jmx_scrape_cached_beans",
-                        Type.GAUGE,
-                        "Number of beans with their matching rule cached",
-                        samples));
-        return mfsList;
-    }
+        jmxScrapeDurationSeconds.set((System.nanoTime() - start) / 1.0E9);
+        jmxScrapeError.set(error);
+        jmxScrapeCachedBeans.set(stalenessTracker.cachedCount());
 
-    public List<MetricFamilySamples> describe() {
-        List<MetricFamilySamples> sampleFamilies = new ArrayList<>();
-        sampleFamilies.add(
-                new MetricFamilySamples(
-                        "jmx_scrape_duration_seconds",
-                        Type.GAUGE,
-                        "Time this JMX scrape took, in seconds.",
-                        new ArrayList<>()));
-        sampleFamilies.add(
-                new MetricFamilySamples(
-                        "jmx_scrape_error",
-                        Type.GAUGE,
-                        "Non-zero if this scrape failed.",
-                        new ArrayList<>()));
-        sampleFamilies.add(
-                new MetricFamilySamples(
-                        "jmx_scrape_cached_beans",
-                        Type.GAUGE,
-                        "Number of beans with their matching rule cached",
-                        new ArrayList<>()));
-        return sampleFamilies;
-    }
+        MetricSnapshots.Builder result = MetricSnapshots.builder();
 
-    /** Convenience function to run standalone. */
-    public static void main(String[] args) throws Exception {
-        String hostPort = "";
-        if (args.length > 0) {
-            hostPort = args[0];
+        for (CounterSnapshot.Builder counter : receiver.countersMap.values()) {
+            result.metricSnapshot(counter.build());
         }
-        JmxCollector jc =
-                new JmxCollector(("{" + "`hostPort`: `" + hostPort + "`," + "}").replace('`', '"'));
-        for (MetricFamilySamples mfs : jc.collect()) {
-            System.out.println(mfs);
+
+        for (GaugeSnapshot.Builder gauge : receiver.gaugeMap.values()) {
+            result.metricSnapshot(gauge.build());
+        }
+
+        for (UnknownSnapshot.Builder unknown : receiver.unknownMap.values()) {
+            result.metricSnapshot(unknown.build());
         }
+
+        return result.build();
+    }
+
+    @Override
+    // TODO remove once bug is fixed in client_java
```

Comment:
Can this be removed? I'm not aware of a bug.

Reply:
Removing the method - this was resolved in 1.1.0 with...

[ENHANCEMENT] Avoid duplicate call to the scrape() method in the default Collector implementation. This does not affect the core metrics, because the core metrics override getPrometheusName(), so they don't use the default implementation.